ntb_hw_switchtec.c

```diff
@@ -992,20 +992,84 @@ static int config_rsvd_lut_win(struct switchtec_ntb *sndev,
 	return 0;
 }
 
-static int config_req_id_table(struct switchtec_ntb *sndev,
-			       struct ntb_ctrl_regs __iomem *mmio_ctrl,
-			       int *req_ids, int count)
+static int add_req_id(struct switchtec_ntb *sndev,
+		      struct ntb_ctrl_regs __iomem *mmio_ctrl, int req_id)
 {
 	int i, rc = 0;
+	int slot = -1;
 	u32 error;
-	u32 proxy_id;
+	int table_size;
+	u32 proxy_id = 0;
+	bool exist = false;
+	bool added = true;
+
+	table_size = ioread16(&mmio_ctrl->req_id_table_size);
+
+	rc = switchtec_ntb_part_op(sndev, mmio_ctrl,
+				   NTB_CTRL_PART_OP_LOCK,
+				   NTB_CTRL_PART_STATUS_LOCKED);
+	if (rc)
+		return rc;
+
+	iowrite32(NTB_PART_CTRL_ID_PROT_DIS, &mmio_ctrl->partition_ctrl);
+
+	for (i = 0; i < table_size; i++) {
+		proxy_id = ioread32(&mmio_ctrl->req_id_table[i]);
 
-	if (ioread16(&mmio_ctrl->req_id_table_size) < count) {
+		if (!(proxy_id & NTB_CTRL_REQ_ID_EN) && (slot == -1))
+			slot = i;
+
+		if ((proxy_id & NTB_CTRL_REQ_ID_EN) &&
+		    (proxy_id >> 16) == req_id) {
+			exist = true;
+			break;
+		}
+	}
+
+	if (!exist) {
+		if (slot == -1) {
+			dev_err(&sndev->stdev->dev,
+				"Not enough requester IDs available.\n");
+			added = false;
+		} else {
+			iowrite32(req_id << 16 | NTB_CTRL_REQ_ID_EN,
+				  &mmio_ctrl->req_id_table[slot]);
+
+			dev_dbg(&sndev->stdev->dev,
+				"Requester ID %02X:%02X.%X -> BB:%02X.%X\n",
+				req_id >> 8, (req_id >> 3) & 0x1F,
+				req_id & 0x7, (proxy_id >> 4) & 0x1F,
+				(proxy_id >> 1) & 0x7);
+			added = true;
+		}
+	}
+
+	rc = switchtec_ntb_part_op(sndev, mmio_ctrl,
+				   NTB_CTRL_PART_OP_CFG,
+				   NTB_CTRL_PART_STATUS_NORMAL);
+
+	if (rc == -EIO) {
+		error = ioread32(&mmio_ctrl->req_id_error);
 		dev_err(&sndev->stdev->dev,
-			"Not enough requester IDs available.\n");
-		return -EFAULT;
+			"Error setting up the requester ID table: %08x\n",
+			error);
 	}
 
+	if (!added)
+		return -EFAULT;
+
+	return 0;
+}
+
+static int cls_req_ids(struct switchtec_ntb *sndev,
```

Comment:
I don't think cls is a common abbreviation.... I assume this is clear? Make it 'clear' or 'clr'.